### PR TITLE
Add restart policy information from API and Kubernetes

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1281654

### CODE REVIEW

**Comment by mburke5678** [comment]
@bfallonf @vikram-redhat 

I added the paragraph on  from the API guide and some info from Kubernetes docs to the Core Concepts -> Pods and Services topic (is there a better place?).

PTAL to see if this answers your concern. If so, I will ask for SME review to make sure I didn't add anything that should not be or need more info.

**Comment by bfallonf** [comment]
@mburke5678 I think this fulfills the BZ. Thanks :wave:

**Comment by mburke5678** [comment]
@mdshuai PTAL

**Comment by mburke5678** [comment]
[rev_history]
|xref:../architecture/core_concepts/pods_and_services.adoc#architecture-core-concepts-pods-and-services[Pods and Services]
|Added a new section on xref:../architecture/core_concepts/pods_and_services.adoc#admin-manage-pod-restart[Pod Restart Policy].
%

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

File: architecture/core_concepts/pods_and_services.adoc [hunk 1]
```diff
@@ -139,12 +139,13 @@ context is very restrictive but administrators can modify this as needed.
 within the container. In this case, there is a volume for storing the registry's
 data, and one for access to credentials the registry needs for making requests
 against the {product-title} API.
-<9> Pods making requests against the {product-title} API is a common enough pattern
+<9> The xref:admin-manage-pod-restart[pod restart policy] with possible values `Always`, `OnFailure`, and `Never`. The default value is Always.
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: Monospace Always the second time it's used.

File: architecture/core_concepts/pods_and_services.adoc [hunk 2]
```diff
@@ -159,6 +160,51 @@ link:https://kubernetes.io/docs/concepts/workloads/pods/pod/[Kubernetes pod docu
 has details about the functionality and purpose of pods.
 ====
 
+[[admin-manage-pod-restart]]
+=== Pod Restart Policy
+//from https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy
+A pod restart policy determines how {product-title} responds when containers in a pod exit.
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: I think there is a word missing here.  Is it supposed to be 'when containers are in a pod exit'?

File: architecture/core_concepts/pods_and_services.adoc [hunk 3]
```diff
@@ -159,6 +160,51 @@ link:https://kubernetes.io/docs/concepts/workloads/pods/pod/[Kubernetes pod docu
...
+The policy applies to all containers in that pod.
+
+The possible values are:
+
+* `Always` - Tries restarting an successfully exited container on the pod continuously, with an exponential back-off delay of (10s, 20s, 40s) until the pod is restarted. The default is `Always`.
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: change 'an' to 'a'

Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: This is awkward:  "delay of (10s, 20s, 40s) until"

File: architecture/core_concepts/pods_and_services.adoc [hunk 4]
```diff
@@ -159,6 +160,51 @@ link:https://kubernetes.io/docs/concepts/workloads/pods/pod/[Kubernetes pod docu
...
+* `OnFailure` - Tries restarting a failed container on the pod with an exponential back-off delay of (10s, 20s, 40s) capped at five minutes.
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: This is awkward: "delay of (10s, 20s, 40s) until"

Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: five should be 5

File: architecture/core_concepts/pods_and_services.adoc [hunk 5]
```diff
@@ -159,6 +160,51 @@ link:https://kubernetes.io/docs/concepts/workloads/pods/pod/[Kubernetes pod docu
...
+* `Never` - Does not try to restart exited or failed containers on the pod. Pods immediately fail and exit.
+
+//https://kubernetes-v1-4.github.io/docs/user-guide/pod-states/
+Once bound to a node, a pod will never be rebound to another node. This means that a controller is necessary in order for a pod to survive node failure:
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: I don't think the word 'be' is necessary.

File: architecture/core_concepts/pods_and_services.adoc [hunk 6]
```diff
@@ -159,6 +160,51 @@ link:https://kubernetes.io/docs/concepts/workloads/pods/pod/[Kubernetes pod docu
...
+|Pods that are expected to not terminate (such as web servers)
+|xref:../../architecture/core_concepts/deployments.adoc#replication-controllers[Replication Controller]
+| `Always`.
+
+|Pods that need to run one-per-machine
+|xref:../../dev_guide/daemonsets.adoc#dev-guide-daemonsets[Daemonset]
+|Any
+|===
+
+If a container on a pod fails and the restart policy is set to `OnFailure`, the pod stays on the node and the container is re-started. If you do not want the container to
```
Comment (nengard) [inline architecture/core_concepts/pods_and_services.adoc]: Remove hyphen from 're-started'

**Review by ncbaratta — CHANGES_REQUESTED** [review]

File: architecture/core_concepts/pods_and_services.adoc (hunk 2 above)
Comment (mburke5678) [reply architecture/core_concepts/pods_and_services.adoc]: _containers in a pod_ and _containers in the pod_ seem to be common phrases, based on a search in Atom. It is like _peas in a pod_!

**Comment by mburke5678** [comment]
Made changes per @nengard review.

**Pull request merged**